lib/models/league_model/league_model.dart

```diff
@@ -0,0 +1,80 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'league_model.freezed.dart';
+part 'league_model.g.dart';
+
+@freezed
+class LeagueModel with _$LeagueModel {
+    const factory LeagueModel({
+        League? league,
+        Country? country,
+        List<Season>? seasons,
+    }) = _LeagueModel;
+
+    factory LeagueModel.fromJson(Map<String, dynamic> json) => _$LeagueModelFromJson(json);
+}
+
+@freezed
+class Country with _$Country {
```

Comment:
Please move the rest of classes to separated files

Reply:
done